src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/ui/classbrowser/HTMLGenerator.java

```diff
@@ -1680,34 +1676,28 @@ public String genHTMLForFieldNames(InstanceKlass klass) {
 
    protected String genHTMLListForFields(InstanceKlass klass) {
       Formatter buf = new Formatter(genHTML);
-      U2Array fields = klass.getFields();
-      int numFields = klass.getAllFieldsCount();
+      sun.jvm.hotspot.oops.Field[] fields = sun.jvm.hotspot.oops.Field.getFields(klass);
+      int numFields = fields.length;
       if (numFields != 0) {
          buf.h3("Fields");
          buf.beginList();
          for (int f = 0; f < numFields; f++) {
-           sun.jvm.hotspot.oops.Field field = klass.getFieldByIndex(f);
-           String f_name = ((NamedFieldIdentifier)field.getID()).getName();
-           Symbol f_sig  = field.getSignature();
-           Symbol f_genSig = field.getGenericSignature();
-           AccessFlags acc = field.getAccessFlagsObj();
-
            buf.beginListItem();
-           buf.append(genFieldModifierString(acc));
+           buf.append(genFieldModifierString(fields[f].getAccessFlagsObj()));
```

Comment:
Was there a reason to not continue to use local variables for the Field data that is fetched? It would make reading the diffs easier.